ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -340,10 +317,61 @@ def create_random_action(self, n_agents: int) -> np.ndarray:
                         size=(n_agents),
                         dtype=np.int32,
                     )
-                    for i in range(self.action_size)
+                    for i in range(self.discrete_size)
                 ]
             )
-            return action
+        return action
+
+    def _validate_action(
+        self, actions: np.ndarray, n_agents: int, name: str
+    ) -> np.ndarray:
+        """
+        Validates that action has the correct action dim
+        for the correct number of agents and ensures the type.
+        """
+        if self.continuous_size > 0:
+            _size = self.continuous_size
+        else:
+            _size = self.discrete_size
+        _expected_shape = (n_agents, _size)
+        if actions.shape != _expected_shape:
+            raise UnityActionException(
+                f"The behavior {name} needs an input of dimension "
+                f"{_expected_shape} for (<number of agents>, <action size>) but "
+                f"received input of dimension {actions.shape}"
+            )
+        _expected_type = np.float32 if self.is_continuous() else np.int32
+        if actions.dtype != _expected_type:
+            actions = actions.astype(_expected_type)
+        return actions
+
+    @staticmethod
+    def create_continuous(continuous_size: int) -> "ActionSpec":
```

Comment:
We have create_random and create_continuous in the API. This is ambiguous because it looks like these 2 methods will return the same thing while they will not. We should rename `create_random` to `create_random_action` or `random_action`.

Reply:
renamed to random_action